From c60b784bdfc4368c8e9fd2f9f233eb74f05453be Mon Sep 17 00:00:00 2001
From: Yumi Yukimura <me.cafebabe@gmail.com>
Date: Tue, 5 Aug 2025 03:50:41 +0800
Subject: [PATCH] msm: Force disable UBWC with `vendor.minigbm.avoid_ubwc` prop

Change-Id: I67cf8eda41358a168a414394b1c8208e974ecec5
---
 msm.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/msm.c b/msm.c
index ee585c7..6a132fb 100644
--- a/msm.c
+++ b/msm.c
@@ -205,6 +205,18 @@ static void msm_add_ubwc_combinations(struct driver *drv, const uint32_t *format
  */
 static bool should_avoid_ubwc(void)
 {
+#ifdef __ANDROID__
+    static bool prop_parsed = false, prop_value = false;
+    const char *prop_buf;
+    if (prop_parsed) {
+        return prop_value;
+    } else {
+        prop_buf = drv_get_os_option("vendor.minigbm.avoid_ubwc");
+        prop_value = prop_buf && !strcmp(prop_buf, "true");
+        prop_parsed = true;
+        return prop_value;
+    }
+#endif
 #ifndef __ANDROID__
 	/* waffle is buggy and, requests a renderable buffer (which on qcom platforms, we
 	 * want to use UBWC), and then passes it to the kernel discarding the modifier.
-- 
2.47.3

