From 9a449babb7e860fa95be8796caccb56d204b1c22 Mon Sep 17 00:00:00 2001
From: hmtheboy154 <buingoc67@gmail.com>
Date: Sat, 23 Nov 2024 07:20:30 -0500
Subject: [PATCH 10/12] fixup! hbm: add a new allocator

Signed-off-by: hmtheboy154 <buingoc67@gmail.com>
---
 hbm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hbm.c b/hbm.c
index 9c6e4be..27693c1 100644
--- a/hbm.c
+++ b/hbm.c
@@ -65,7 +65,7 @@ struct hbm_resource {
 
 static uint32_t pick_staging_memory_type(struct hbm *hbm);
 
-static void hbm_log(int32_t lv, const char *msg, void *data)
+static void hbm_log(enum hbm_log_level lv, const char *msg, void *data)
 {
 	enum drv_log_level drv_lv;
 	switch (lv) {
-- 
2.39.5

